new-log-viewer/src/components/modals/SettingsModal/ConfigDialog.tsx

```diff
@@ -0,0 +1,209 @@
+        label: "Decoder: Format string",
+        name: LOCAL_STORAGE_KEY.DECODER_OPTIONS_FORMAT_STRING,
+        type: "text",
+    },
+    {
+        helperText: "[JSON] Key to extract the log level from.",
+        initialValue: getConfig(CONFIG_KEY.DECODER_OPTIONS).logLevelKey,
+        label: "Decoder: Log level key",
+        name: LOCAL_STORAGE_KEY.DECODER_OPTIONS_LOG_LEVEL_KEY,
+        type: "text",
+    },
+    {
+        helperText: "[JSON] Key to extract the log timestamp from.",
+        initialValue: getConfig(CONFIG_KEY.DECODER_OPTIONS).timestampKey,
+        label: "Decoder: Timestamp key",
+        name: LOCAL_STORAGE_KEY.DECODER_OPTIONS_TIMESTAMP_KEY,
+        type: "text",
+    },
+    {
+        helperText: "Number of log messages to display per page.",
+        initialValue: getConfig(CONFIG_KEY.PAGE_SIZE),
+        label: "View: Page size",
+        name: LOCAL_STORAGE_KEY.PAGE_SIZE,
+        type: "number",
+    },
+];
+
+/**
+ * Handles the reset event for the configuration form.
+ *
+ * @param ev The form event triggered by the reset action.
+ */
+const handleConfigFormReset = (ev: React.FormEvent) => {
+    ev.preventDefault();
+    window.localStorage.clear();
+    window.location.reload();
+};
+
+/**
+ * Handles the submit event for the configuration form.
+ *
+ * @param ev The form event triggered by the submit action.
+ */
+const handleConfigFormSubmit = (ev: React.FormEvent) => {
+    ev.preventDefault();
+    const formData = new FormData(ev.target as HTMLFormElement);
+    const getFormDataValue = (key: string) => formData.get(key) as string;
+
+    const formatString = getFormDataValue(LOCAL_STORAGE_KEY.DECODER_OPTIONS_FORMAT_STRING);
+    const logLevelKey = getFormDataValue(LOCAL_STORAGE_KEY.DECODER_OPTIONS_LOG_LEVEL_KEY);
+    const timestampKey = getFormDataValue(LOCAL_STORAGE_KEY.DECODER_OPTIONS_TIMESTAMP_KEY);
+    const pageSize = getFormDataValue(LOCAL_STORAGE_KEY.PAGE_SIZE);
+
+    let error: Nullable<string> = null;
+    error ||= setConfig({
+        key: CONFIG_KEY.DECODER_OPTIONS,
+        value: {formatString, logLevelKey, timestampKey},
+    });
+    error ||= setConfig({
+        key: CONFIG_KEY.PAGE_SIZE,
+        value: Number(pageSize),
+    });
+
+    if (null !== error) {
+        // eslint-disable-next-line no-warning-comments
+        // TODO: Show an error pop-up once NotificationProvider is implemented.
+        // eslint-disable-next-line no-alert
+        window.alert(error);
+    } else {
+        window.location.reload();
+    }
+};
+
+/**
+ * Renders a configuration settings form.
+ *
+ * @return
+ */
+const ConfigDialog = forwardRef<HTMLFormElement>((_, ref) => {
+    const {setMode, mode} = useColorScheme();
+
+    return (
+        <form
+            ref={ref}
+            tabIndex={-1}
+            onReset={handleConfigFormReset}
+            onSubmit={handleConfigFormSubmit}
+        >
+            <ModalDialog
+                maxWidth={"lg"}
```

Comment:
```suggestion
                minWidth={"md"}
```

Reply:
@Henry8192 Shall we apply this? Currently the modal window is too narrow that the Format String input box cannot be shown fully.
![image](https://github.com/user-attachments/assets/7ed4864e-b6ac-4747-b62b-a45f25b54280)

Reply:
Yup let's do this.